openforcefield/utils/toolkits.py

```diff
@@ -3438,6 +3442,360 @@ def assign_partial_charges(
+
+        Notes
+        -----
+        * The partial charges will be taken from the molecule directly.
+
+        Parameters
+        ----------
+        molecule
+            The molecule containing the partial charges.
+        conformer
+            The conformer to compute the energy of. This should be a unit wrapped
+            numpy array with shape=(n_atoms, 3) with units compatible with angstroms.
+
+        Returns
+        -------
+            The electrostatic interaction energy in units of [e^2 / Angstrom].
+        """
+
+        if molecule.partial_charges is None:
+            raise ValueError("The molecule has no partial charges assigned.")
+
+        partial_charges = np.abs(
+            molecule.partial_charges.value_in_unit(unit.elementary_charge)
+        ).reshape(-1, 1)
+
+        # Build an exclusion list for 1-2 and 1-3 interactions.
+        excluded_pairs = {
+            *[(bond.atom1_index, bond.atom2_index) for bond in molecule.bonds],
+            *[
+                (angle[0].molecule_atom_index, angle[-1].molecule_atom_index)
+                for angle in molecule.angles
+            ],
+        }
+
+        # Build the distance matrix between all pairs of atoms.
+        coordinates = conformer.value_in_unit(unit.angstrom)
+
+        distances = np.sqrt(
+            np.sum(np.square(coordinates)[:, np.newaxis, :], axis=2)
+            - 2 * coordinates.dot(coordinates.T)
+            + np.sum(np.square(coordinates), axis=1)
+        )
+        # Handle edge cases where the squared distance is slightly negative due to
+        # precision issues
+        np.fill_diagonal(distances, 0.0)
+
+        inverse_distances = np.reciprocal(
+            distances, out=np.zeros_like(distances), where=~np.isclose(distances, 0.0)
+        )
+
+        # Multiply by the charge products.
+        charge_products = partial_charges @ partial_charges.T
+
+        for x, y in excluded_pairs:
+            charge_products[x, y] = 0.0
+            charge_products[y, x] = 0.0
+
+        interaction_energies = inverse_distances * charge_products
+
+        return 0.5 * interaction_energies.sum()
+
+    @classmethod
+    def _elf_compute_rms_matrix(cls, molecule: "Molecule") -> np.ndarray:
+        """Computes the symmetric RMS matrix of all conformers in a molecule.
+
+        Parameters
+        ----------
+        molecule
+            The molecule containing the conformers.
+
+        Returns
+        -------
+            The RMS matrix with shape=(n_conformers, n_conformers).
+        """
+
+        from rdkit import Chem
+        from rdkit.Chem import AllChem
+
+        rdkit_molecule: Chem.RWMol = molecule.to_rdkit()
+        n_conformers = len(molecule.conformers)
+
+        rms_index = np.tril_indices(n_conformers, k=-1, m=n_conformers)
+        rms_matrix = np.zeros((n_conformers, n_conformers))
+
+        rms_matrix[rms_index] = np.array(AllChem.GetConformerRMSMatrix(rdkit_molecule))
```

Comment:
Unfortunately ``GetConformerRMSMatrix`` won't handle automorphs. I wasn't sure whether to use ``GetBestRMS`` as, from the docs:

> This function will attempt to align all permutations of matching atom orders in both molecules, for some molecules it will lead to ‘combinatorial explosion’ especially if hydrogens are present. Use ‘rdkit.Chem.AllChem.AlignMol’ to align molecules without changing the atom order.

i.e. this could become a bit of a performance hole for more complex molecules. 

I'm not sure how common it is for conformer generators to produce automorphs, so we may be safe to not check for them here? I'll leave it up to you @j-wags to decide whether you're ok with the possible combinatorial explosion here.

Reply:
Let's play it safe for now and use GetBestRMS -- Given that this will probably be coupled with a bunch of AM1 calculations, the runtime that this adds to the whole operation shouldn't be too high. We can re-evaluate this choice if we start moving toward using a fast ML charge assignment method and performance becomes an issue.

Reply:
Ok sure - this should be resolved by 704ac1d